dbt/adapters/databricks/connections.py

```diff
@@ -213,8 +213,6 @@ def _execute_cursor(
         return self.get_result_from_cursor(cursor)
 
     def list_schemas(self, database: Optional[str], schema: Optional[str] = None) -> Table:
-        database = database if isinstance(database, str) else None
```

Comment:
So previously we only handled the `Undefined` values for `list_schemas` but now we handle this for all available methods in the adapter class.